src/main/java/de/thomas_oster/liblasercut/drivers/Ruida.java

```diff
@@ -0,0 +1,1406 @@
+  private static final long[] EngraveMaxVelocity = {800,800,2048};
+  private static final long[] VectorAcceleration = {100000,25000,20000};
+  private static final long[] VectorMaxVelocity = {1000,1000,1000};
+//  private static final byte FlipLaserPWMPower = 1;
+//  private static final byte FlipLaserOutput = 0;
+
+//  private static final byte HomeDirection = 1;
+//  private static final byte[] FlipHomeDirection = {1,0,0};
+//  private static final byte[] LimitContCondition = {0,0,0,0};
+//  private static final long[] MaxSteps = {250,500,500};
+//  private static final long[] TableSize = {20000,12000,30000};
+
+  protected static final String[] uploadMethodList = {UPLOAD_METHOD_FILE, UPLOAD_METHOD_IP, UPLOAD_METHOD_SERIAL};
+  public static final int SOURCE_PORT = 40200; // used by rdworks in Windows
+  public static final int DEST_PORT = 50200; // fixed UDP port
+
+  private ByteStream stream;
+  private transient InputStreamReader in;
+  private transient PrintStream out;
+  private transient DatagramSocket socket;
+  private transient InetAddress address;
+  private transient CommPort port;
+  private transient CommPortIdentifier portIdentifier;
+
+  private float focus2mm(int focus)
+  {
+    return (float) (focus * FOCUSWIDTH);
+  }
+
+  protected int baudRate = 921600;
+
+  public int getBaudRate()
+  {
+    return baudRate;
+  }
+
+  public void setBaudRate(int baudRate)
+  {
+    this.baudRate = baudRate;
+  }
+
+  /**
+   * Time to wait before firsts reads of serial port.
+   * See autoreset feature on arduinos.
+   */
+  protected int initDelay = 5;
+
+  public int getInitDelay()
+  {
+    return initDelay;
+  }
+
+  public void setInitDelay(int initDelay)
+  {
+    this.initDelay = initDelay;
+  }
+
+  protected String host = "192.168.1.1";
+
+  public String getHost()
+  {
+    return host;
+  }
+
+  public void setHost(String host)
+  {
+    this.host = host;
+  }
+
+  protected String comport = "auto";
+
+  public String getComport()
+  {
+    return comport;
+  }
+
+  public void setComport(String comport)
+  {
+    this.comport = comport;
+  }
+
+  /* -----------------------------------------------------------------------*/
+
+  public Ruida()
+  {
+  }
+
+  /**
+   * Copies the current instance with all config settings, because
+   * it is used for save- and restoring
+   * @return
+   */
+  @Override
+  public Ruida clone() {
+    Ruida clone = new Ruida();
+    clone.copyProperties(this);
+    return clone;
+  }
+
+  @Override
+  public LaserProperty getLaserPropertyForVectorPart()
+  {
+    return new FloatMinMaxPowerSpeedFrequencyProperty();
+  }
+
+  @Override
+  public LaserProperty getLaserPropertyForRasterPart()
+  {
+    return new FloatMinMaxPowerSpeedFrequencyProperty();
+  }
+
+  @Override
+  public LaserProperty getLaserPropertyForRaster3dPart()
+  {
+    return new FloatMinMaxPowerSpeedFrequencyProperty();
+  }
+
+  @Override
+  public boolean canEstimateJobDuration()
+  {
+    return true;
```

Comment:
should not be true, there is no corresponding estimateJobDuration(LaserJob) method. If you want to implement that, then see estimateJobDuration() in the GenericGcodeCutter and EpilogCutter class for an example, you just need to adapt the constants.

Reply:
set to `false` for now